Add dedicated query for assessment form definition lookup

gigxz opened the pull request:
## Description

Add a dedicated query for looking up the form definition to use for an assessment.

This is needed to support multiple custom assessments https://github.com/greenriver/hmis-frontend/pull/619

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

gigxz committed:
Add dedicated query for assessment form definition lookup

gigxz committed:
Comments

gigxz committed:
Comments

gigxz committed:
Add test cases for project rule application

gigxz committed:
Update other lookup test

gigxz committed:
Merge branch 'release-101' into gig/assessment-form-definition-lookup

gigxz committed:
safe accessor

gigxz committed:
Merge branch 'release-101' into gig/assessment-form-definition-lookup

ttoomey commented on drivers/hmis/app/graphql/types/hmis_schema/query_type.rb:
```diff
@@ -171,18 +171,44 @@ def file(id:)
+      argument :project_id, ID, required: true, description: 'Project to use for Rule filtering on the definition'
+      argument :id, ID, required: false, description: 'Form Definition ID, if known'
+      argument :role, Types::Forms::Enums::AssessmentRole, required: false, description: 'Assessment role, if looking up by role'
+      argument :assessment_date, GraphQL::Types::ISO8601Date, required: false, description: 'Date to use for Rule filtering on the definition'
```
maybe validate that we only get one arg?
```suggestion
      argument :assessment_date, GraphQL::Types::ISO8601Date, required: false, description: 'Date to use for Rule filtering on the definition'
        validates required: { one_of: [:id, :role] }

```

ttoomey commented on drivers/hmis/app/graphql/types/hmis_schema/query_type.rb:
```diff
@@ -171,18 +171,44 @@ def file(id:)
-      argument :enrollment_id, ID, required: false
       argument :project_id, ID, required: false
     end
+    def get_form_definition(role:, project_id: nil)
```
do you think it's worth it to rename this? Maybe `record_form_definition` ?

gigxz replied:
sure we might as well

ttoomey reviewed (approved):
Looks good!

gigxz committed:
Merge branch 'release-101' into gig/assessment-form-definition-lookup

gigxz committed:
Add one of validation

gigxz committed:
Consistent query naming

gigxz commented:
Addressed feedback with https://github.com/greenriver/hmis-warehouse/pull/3948/commits/f3de73998f265193fe538af4a80881c6d06757e4 + added separate Enum for the `recordFormDefinition` arg, plus descriptions for each role enum

gigxz committed:
Include assessment and form def IDs in reminders (#3951)

* Update to reminder assessment links

* Update to reminder assessment links

* Address feedback

gigxz committed:
Fix typo on Update role

gigxz committed:
Fix incorrect test assertions
